### PR TITLE
[CustomHandler]Include HttpResponseMessage if fails to deserialize to invocation result

### LINKED ISSUE #6640: [V2][Custom handler] Log response if handler returns non-JSON response
Port fix for #6607

### PR DESCRIPTION
**Pull request by pragnagopa**
Resolves #6640

Port commit # b0eccda4f1323bd7855ff1e5d15491902d3ac47c

### CODE REVIEW

**Review by yojagad — DISMISSED** [review]

File: src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs [hunk 1]
```diff
@@ -114,30 +113,33 @@ private async Task<HttpResponseMessage> SendInvocationRequestAsync(ScriptInvocat
...
+        private static async Task<string> GetHttpContentAsString(HttpContent content, ILogger logger = null)
         {
             if (content != null)
             {
                 bool isMediaTypeOctetOrMultipart = content.Headers != null && Utility.IsMediaTypeOctetOrMultipart(content.Headers.ContentType);
                 // do not log binary data as string
                 if (!isMediaTypeOctetOrMultipart)
                 {
-                    logger.LogTrace(await content.ReadAsStringAsync());
+                    string stringCotent = await content.ReadAsStringAsync();
```
Comment (mathewc) [inline src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs]: spelling errors

File: src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs [hunk 2]
```diff
@@ -114,30 +113,33 @@ private async Task<HttpResponseMessage> SendInvocationRequestAsync(ScriptInvocat
...
+                    logger?.LogTrace(stringCotent);
```
Comment (mathewc) [inline src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs]: Where is this log going? It won't go to system logs, right?

File: src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs (hunk 1 above)
Comment (mathewc) [reply src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs]: Also, this PR is open, but looks like it was checked in?

File: src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs (hunk 2 above)
Comment (pragnagopa) [reply src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs]: This is user logger. When running locally, request/response logged to console.

File: src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs (hunk 1 above)
Comment (pragnagopa) [reply src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs]: Fixed typo. Thanks! This PR is to Port fix to V2. V3 PR is merged.

File: release_notes.md [hunk 3]
```diff
@@ -6,14 +6,22 @@
...
-`{
-"statusCode" : "",
-"status" : "",
-"body": "",
-"headers" : {}
-}`
-Exception is thrown if HttpOutputBindingResponse is not valid Json.
+- **Breaking Changes in CustomHandler**
+    -  Issue #6644 : If enableForwardingHttpRequest is false, http output binding response is expected to be a valid Json object with following optional fields :
+    ```json
```
Comment (pragnagopa) [inline release_notes.md:15]: @anthonychu - updated release notes in V2 as well. Thanks!

**Review by yojagad — APPROVED** [review]

**Pull request merged**